browser/mapping.go

```diff
@@ -708,6 +743,54 @@ func mapBrowser(vu moduleVU, b api.Browser, wsURL string, isRemoteBrowser bool)
 	}
 }
 
+// getOrInitBrowser retrieves the browser for the iteration from the browser registry
+// if it is already initialized. Otherwise initializes a new browser for the iteration
+// and stores it in the registry.
+func getOrInitBrowser(
+	bt *chromium.BrowserType, vu moduleVU, wsURL string, isRemoteBrowser bool,
+) (api.Browser, error) {
+	// Index browser pool per VU-scenario-iteration
+	id := fmt.Sprintf("%d-%s-%d",
+		vu.State().VUID,
+		k6ext.GetScenarioName(vu.Context()),
+		vu.State().Iteration,
+	)
+
+	var (
+		ok  bool
+		err error
+		b   api.Browser
+	)
+
+	if b, ok = vu.getBrowser(id); ok {
+		return b, nil
+	}
+
+	if isRemoteBrowser {
+		b, err = bt.Connect(wsURL)
+		if err != nil {
+			return nil, err //nolint:wrapcheck
+		}
+	} else {
+		var pid int
+		b, pid, err = bt.Launch()
+		if err != nil {
+			return nil, err //nolint:wrapcheck
+		}
+		vu.registerPid(pid)
+	}
+
+	vu.setBrowser(id, b)
+
+	go func(ctx context.Context) {
```

Comment:
IDK if it's possible, but can we pass the context to `bt.Connect` and `bt.Launch` so that we don't have to track it in a new goroutine here? Launching a goroutine also pollutes the logs and makes it hard to track.

Reply:
We could potentially wait for the VU `context`'s "done" in `bt.Launch` and `bt.Connect` implementations, but, because we have to call `vu.deleteBrowser` once the browser is closed, we would have to export that method and also the iteration ID generation "logic" used as index for the registry to `chromium` pkg, which does not seem ideal either.

What is your take on this?

In any case this goroutine will be removed once we have a reliable event from k6 to wait for at the end of the iteration.

Reply:
Since this is will be removed soon, I don't mind keeping it as it is 👍 Creating an issue about this would help, thanks!